lib/cocoapods-core/cdn_source.rb

```diff
@@ -68,6 +69,21 @@ def preheat_existing_files
       end
     end
 
+    def files_definitely_to_update
+      [repo.join('**/*.yml'), repo.join('**/*.txt')].
+        map(&Pathname.method(:glob)).
+        flatten.
+        map { |f| f.relative_path_from(repo).to_s }
+    end
+
+    def deprecated_local_podspecs
+      download_file('deprecated_podspecs.txt')
```

Comment:
Is there anyway to short circuit here if `deprecated_podspecs.txt` wasn't changed? Etag matched?

Reply:
All downloads in update mode include the `if-none-match` header.

Reply:
But do we always check all deprecated local podspecs? If `deprecated_podspecs.txt` `304`s, couldn't we just return `[]` here? Since deprecations are rare, this could be a big win.

Reply:
I don't quite follow what you mean. This method intersects the user's repo directory with the entire list of deprecations.

Reply:
So if I understand what this does, it downloads `deprecated_podspecs.txt` if needed.
Then finds all locally tracked deprecated podspecs and calls `download_file` for them. Updating if needed (not 304), but a request is still made.

My thinking was if `deprecated_podspecs.txt` 304ed, then there is no reason to check all the locally known deprecated_podspecs for changes. In which case we can lower the network calls made even more.

Reply:
Also I don't know how much `Core` knows about parsing podspecs, but taking a few lines from your Specs script update:
```
podspec = Pod::Specification.from_file(podspec_path)
podspec.deprecated?
```
We only need to make a request for locally tracked deprecated podspecs where the above is `false`.

Reply:
Theoretically, what you're saying sounds correct.  
However, I chose not to do this at this time since the expected value of these unnecessary updates is quite tiny. Deprecated podspecs are less than 2% of the total, and because they are deprecated, the chances of them being downloaded are even smaller.

The additional code complexity just doesn't seem justified.